File: Cap 09/Topico 9.4/Topico 9.4.1 a 9.4.3 juntos/regmatrix.cpp
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topicos 9.4.1 a 9.4.3
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ
 * 
 */

#include <iostream>
#include <string>
#include <iomanip>

using namespace std;

typedef float BIMESTRE[4];

struct CAD_ALUNO {
  char NOME[256];
  char TURMA;
  int SALA;
  BIMESTRE NOTAS;
};

CAD_ALUNO ALUNO[8];
int I, J;

int main(void) {
	
  cout << "REGISTRO ESCOLAR - MATRICIAL (8 ALUNOS)\n\n";
  for (I = 0; I < 8; I++) {
    cout << "ALUNO " << I + 1 << endl;
    cout << "Entre o nome ......: ";
    cin.getline(ALUNO[I].NOME, sizeof(ALUNO[I].NOME));
    cout << "Entre a turma .....: ";
    cin >> ALUNO[I].TURMA;
    cin.ignore();
    cout << "Entre a sala ......: ";
    cin >> ALUNO[I].SALA;
    cin.ignore();
    for (J = 0; J < 4; J++) {
      cout << "Entre a " << J + 1 << "a. nota ..: ";
      cin >> ALUNO[I].NOTAS[J];
      cin.ignore();
    }
    cout << endl;
  }

  cout << "DADOS DOS ALUNOS" << endl;
  cout << setw(5) << "Aluno ";
  cout << setw(30) << left << "Nome                           ";
  cout << setw(4) << "Sala ";
  cout << setw(5) << "Nota1 ";
  cout << setw(5) << "Nota2 ";
  cout << setw(5) << "Nota3 ";
  cout << setw(5) << "Nota4" << endl;
  cout << "----- ";
  cout << "------------------------------ ";
  cout << "---- ";
  cout << "----- ";
  cout << "----- ";
  cout << "----- ";
  cout << "-----" << endl;
  for (I = 0; I < 8; I++) {
    cout << setw(5)  << right << I + 1 << " ";
    cout << setw(30) << left << string(ALUNO[I].NOME).substr(0, 30) << " ";
    cout << setw(4)  << right << ALUNO[I].SALA << " ";
    for (J = 0; J < 4; J++) {
      cout << setw(5) << fixed << setprecision(1) << ALUNO[I].NOTAS[J] << " ";
    }
    cout << endl;
  }

  return 0;
}

File: Cap 09/Topico 9.4/Topico 9.4.1 a 9.4.3 juntos/regmatrix.pl
=pod

 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 9 - Topicos 9.4.1 a 9.4.3
 * 
 * Programa ..: MANIPULACAO DE REGISTRO DE MATRIZ

=cut

use strict;
use warnings;
use feature 'say';

use constant NUM_ALUNOS => 8;

my @ALUNO;
my $I;
my $J;

say "REGISTRO ESCOLAR - MATRICIAL (8 ALUNOS)\n";

for ($I = 0; $I < NUM_ALUNOS; $I++) {
  say "ALUNO ", $I + 1;
  
  print "Entre o nome ......: ";
  $ALUNO[$I]{'NOME'} = <STDIN>;
  chomp $ALUNO[$I]{'NOME'};

  print "Entre a turma .....: ";
  $ALUNO[$I]{'TURMA'} = <STDIN>;
  chomp $ALUNO[$I]{'TURMA'};

  print "Entre a sala ......: ";
  $ALUNO[$I]{'SALA'} = <STDIN>;
  chomp $ALUNO[$I]{'SALA'};

  for ($J = 0; $J < 4; $J++) {
    print "Entre a ", $J + 1, "a. nota ..: ";
    $ALUNO[$I]{'NOTAS'}[$J] = <STDIN>;
    chomp $ALUNO[$I]{'NOTAS'}[$J];
  }

  say "";
}

say "DADOS DOS ALUNOS";
say sprintf("%5s %-30s %4s %5s %5s %5s %5s", "Aluno", "Nome", "Sala", "Nota1", "Nota2", "Nota3", "Nota4");
say "----- ------------------------------ ---- ----- ----- ----- -----";
for ($I = 0; $I < NUM_ALUNOS; $I++) {
  say sprintf("%5d %-30s %4d %5s %5s %5s %5s",
    $I + 1,
    substr($ALUNO[$I]{'NOME'}, 0, 30),
    $ALUNO[$I]{'SALA'},
    sprintf("%.1f", $ALUNO[$I]{'NOTAS'}[0]),
    sprintf("%.1f", $ALUNO[$I]{'NOTAS'}[1]),
    sprintf("%.1f", $ALUNO[$I]{'NOTAS'}[2]),
    sprintf("%.1f", $ALUNO[$I]{'NOTAS'}[3])
  );
}

